HIP/hip-external-transaction-signing.md

```diff
@@ -0,0 +1,337 @@
+protocol. Some additional definitions of hedera signing operations is needed to
+finish our the full loop.
+
+## User stories
+
+As an SDK user I want to be able to configure my SDK builds to use an external
+signing tool to sign transactions.
+
+As an application I want to be able to use external signing tools at build time
+and at runtime.
+
+## Specification
+
+A websocket based proxy server adhering to CAIP-25 and CAIP-27 JSON-RPC calls
+will be used to provide signing services. Clients will need to configure and
+secure access to this websocket service and then use the provided handshake and
+provider request APIs to sign their transactions. Clients will then either need
+to submit the signed transaction or validate the submitted transaction was
+successfully submitted by the service.
+
+### Signing Proxy Server (CAIP-25 and CAIP-27)
+
+Signing will be supported by an external web socket server. This can either be
+the web server of the hardware wallet or external service, or it may be a bridge
+between hedera and the specific APIs of the hardware wallet or signing service.
+The specific implementation is a detail outside the scope of this spec.
+
+The APIs that will be used will be the CAIP-25 handshake request, the CAIP-27
+provider request wrapping the hedera `sign_transaction` or `send_transaction`
+call specified in this HIP.
+
+#### Sample call loop
+
+To begin a signing session the SDK or app will initiate a web socket connection
+to the proxy. A CAIP-25 handshake call must be successfully made at that point.
+Zero or more signing requests can be made via the CAIP-27 provider request. The
+session is ended by closing the websocket connection.
+
+If the app wishes to change the signing key then a separate websocket connection
+to the proxy server will need to be maintained in parallel or serialized.
+
+### Hedera signing JSON-RPC calls
+
+Two JSON-RPC methods are defined for use in the CAIP-27 provider
+request, `hedera_signTransaction` and `hedera_sendTransaction`.
+
+#### `hedera_signTransaction`
+
+This JSON-RPC method signs a transaction that can be submitted to the network
+via the exiting SDK methods. It is expected there will be some operation on the
+other end to validate the signature, possibly including user approval.
+
+There is one required and one optional parameter: The required parameter
+is `transaction`, which will be the hex encoded protobuf message that is being
+signed. The optional parameter is `pubKey`, which is the public key of the
+desired signature. If the signer has more than one key this is a required field.
+If `pubKey` is specified the signature returned MUST be that of the specified
+key.
+
+> An example not specifying a specific key
+> ```json
+> {
+>     "id": 1,
+>     "jsonrpc": "2.0",
+>     "method": "hedera_signTransaction",
+>     "params": {
+>         "transaction": "fedbca9876543210"
+>     }
+> }
+> ```
+
+> An example specifying a private key
```

Comment:
Probably safer to say you specify a `public` key here, not `private`

Reply:
done